src/style.css

```diff
@@ -0,0 +1,120 @@
+body {
+  margin: 0;
+  font-family: Roboto, sans-serif;
+
+  --blue-color: #00acdc;
+}
+
+* {
+  box-sizing: border-box;
+}
+
+.product-card {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  padding: 29px 14px 14px;
+  border: 1px solid #f3f3f3;
+  border-radius: 5px;
+  width: 200px;
+}
+
+.product-card:hover {
+  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.07);
+}
+
+.product-card__image {
+  margin-bottom: 45px;
+  width: 160px;
+  height: 134px;
+}
+
+.product-card__text {
+  margin-bottom: 6px;
+  font-weight: 500;
+  font-size: 12px;
+  background-color: #fff;
+  line-height: 15px;
+}
+
+.product-card__product-code {
+  width: 100%;
+  font-weight: 400;
+  font-size: 11px;
+  margin-bottom: 15px;
+}
+
+.product-card__rating {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  width: 100%;
+  margin-bottom: 20px;
+}
+
+.stars {
+  display: inline-flex;
+}
+
+.stars__star {
+  width: 15px;
+  height: 15px;
+  background-image: url(images/star.svg);
+  background-repeat: no-repeat;
+  background-position: center;
+}
+
+.stars--1 :nth-child(-n + 1),
+.stars--2 :nth-child(-n + 2),
+.stars--3 :nth-child(-n + 3),
+.stars--4 :nth-child(-n + 4),
+.stars--5 :nth-child(-n + 5) {
+  background-image: url(images/star-active.svg);
+}
+
+.stars__star:not(:nth-child(5)) {
+  margin-right: 4px;
+}
+
+.product-card__rewiews {
+  display: inline-block;
+  font-weight: 400;
+  font-size: 11px;
+}
+
+.product-card__price {
+  display: flex;
+  align-items: baseline;
+  justify-content: space-between;
+  width: 100%;
+  margin-bottom: 20px;
+}
+
+.product-card__price-text {
+  font-weight: 400;
+  font-size: 11px;
+}
+
+.product-card__price-value {
+  font-weight: 700;
+  font-size: 16px;
+}
+
+.product-card__buy-link {
+  display: flex;
+  justify-content: center;
+  padding: 11px 0;
+  width: 100%;
+  background-color: var(--blue-color);
+  border-radius: 5px;
+  color: white;
+  font-weight: 700;
+  font-size: 14px;
+  text-decoration: none;
+}
+
+.product-card__buy-link:hover {
+  background-color: #fff;
+  color: var(--blue-color);
+  border: 1px solid var(--blue-color);
```

Comment:
at this moment when you hovered on your link - the border appears and adds 1px, because of box-sizing: border-box; width and height of your button is recalculating, and text could change its position, so it's better if you write border property inside the button.